test/src/kbasesearchengine/test/main/ObjectDecoratorTest.java

```diff
@@ -0,0 +1,361 @@
+import kbasesearchengine.authorization.AccessGroupProvider;
+import kbasesearchengine.common.FileUtil;
+import kbasesearchengine.events.exceptions.IndexingException;
+import kbasesearchengine.events.exceptions.RetriableIndexingException;
+import kbasesearchengine.main.*;
+import org.apache.commons.io.FileUtils;
+import org.apache.http.HttpHost;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoDatabase;
+
+import kbasesearchengine.common.GUID;
+import kbasesearchengine.events.handler.CloneableWorkspaceClientImpl;
+import kbasesearchengine.events.handler.WorkspaceEventHandler;
+import kbasesearchengine.events.storage.MongoDBStatusEventStorage;
+import kbasesearchengine.events.storage.StatusEventStorage;
+import kbasesearchengine.search.ElasticIndexingStorage;
+import kbasesearchengine.search.IndexingStorage;
+import kbasesearchengine.system.FileLister;
+import kbasesearchengine.system.ObjectTypeParsingRulesFileParser;
+import kbasesearchengine.system.TypeFileStorage;
+import kbasesearchengine.system.TypeMappingParser;
+import kbasesearchengine.system.TypeStorage;
+import kbasesearchengine.system.YAMLTypeMappingParser;
+import kbasesearchengine.test.common.TestCommon;
+import kbasesearchengine.test.controllers.elasticsearch.ElasticSearchController;
+import kbasesearchengine.test.controllers.workspace.WorkspaceController;
+import kbasesearchengine.test.data.TestDataLoader;
+import kbasesearchengine.authorization.AccessGroupCache;
+import kbasesearchengine.authorization.WorkspaceAccessGroupProvider;
+import us.kbase.auth.AuthToken;
+import us.kbase.common.service.JsonClientException;
+import us.kbase.common.service.Tuple11;
+import us.kbase.common.service.Tuple5;
+import us.kbase.common.service.UObject;
+import us.kbase.common.test.controllers.mongo.MongoController;
+import us.kbase.test.auth2.authcontroller.AuthController;
+import workspace.*;
+
+public class ObjectDecoratorTest {
+
+    // should add a setting in the worker and coordinator to shorten the wait time for testing
+    // purposes
+
+    private static AuthController auth = null;
+    private static IndexerWorker worker = null;
+    private static IndexerCoordinator coord = null;
+    private static MongoController mongo;
+    private static MongoClient mc;
+    private static MongoDatabase db;
+    private static ElasticSearchController es;
+    private static IndexingStorage indexStorage;
+    private static WorkspaceController ws;
+    private static WorkspaceEventHandler weh;
+    private static MongoDatabase wsdb;
+    private static WorkspaceClient wsCli1, wsClient;
+    private static AuthToken userToken;
+
+    private static Path tempDirPath;
+
+    private static SearchInterface search;
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```

Comment:
Rather than doing all this, use mocks for SearchMethods and the WorkspaceClient.